solutions/src/main/scala/org/scalalabs/basic/lab01/OOExercise.scala

```diff
@@ -27,10 +27,7 @@ object Euro {
   implicit class EuroInt(val i: Int) extends AnyVal {
     def *(euro: Euro) = euro * i
   }
-
-  //implicit def fromDollar(dollar: Dollar): Euro = Euro.fromCents((DefaultCurrencyConverter.toEuroCents(dollar.inCents)).toInt)
-
-  implicit def fromDollar(dollar: Dollar)(implicit converter: CurrencyConverter): Euro = Euro.fromCents(converter.toEuroCents(dollar.inCents))
+  implicit def fromDollar(dollar: Dollar)(implicit converter: CurrencyConverter = DefaultCurrencyConverter): Euro = Euro.fromCents(converter.toEuroCents(dollar.inCents))
```

Comment:
The first, commented-out solution is the solution to Exercise 4.

The second, active solution is the solution to Exercise 5.

Indeed you cannot have the solutions to exercise 4 and 5 active at the same time. However, I do think it is useful to include both of them in the solutions. Perhaps we should add comments making it more clear what's going on?

Reply:
I know this is the case, but I think it helps if you provide one solution that makes all tests run at once without commenting and uncommenting the one or the other?

What about leaving the unified version as solution and add the others commented with explicit comments?

Reply:
I changed it please have a look...or dont you like the unified solution at all? ;)

Reply:
Hmm, I see what you mean. Indeed it would be good not to have any failing tests in the solutions. On the other hand, I'd rather avoid showing a method with an implicit parameter with a default value: that is complicated for students at this stage, and I would recommend against it in general.

Perhaps we should put the test for either exercise 4 or exercise 5 on 'ignore', and add a comment explaining that to run the other one you also need to select the other solution?

Reply:
right okay. I brought it now back basically to original version but added some more comments to make it clear what is happening. Please check